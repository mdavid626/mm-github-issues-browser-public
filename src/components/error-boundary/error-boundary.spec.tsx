import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ErrorBoundary from './error-boundary';

describe('error-boundary', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockReturnValue(undefined);
  });
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    const { asFragment } = render(
      <ErrorBoundary>
        <div>content</div>
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(console.error).not.toHaveBeenCalled();
  });

  it('should show error page', () => {
    const Component = () => {
      throw new Error('test error');
    };
    const { asFragment } = render(
      <ErrorBoundary>
        <Component />
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(console.error).toHaveBeenCalled();
  });
});
