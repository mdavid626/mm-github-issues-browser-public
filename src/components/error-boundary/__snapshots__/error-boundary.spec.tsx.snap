// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error-boundary should render 1`] = `
<DocumentFragment>
  <div>
    content
  </div>
</DocumentFragment>
`;

exports[`error-boundary should show error page 1`] = `
<DocumentFragment>
  <div
    class="ErrorBoundary"
  >
    <div
      class="ErrorBoundary-title"
    >
      GitHub Issues Browser
    </div>
    <div
      class="ErrorBoundary-description"
    >
      Oops, that's an error
    </div>
  </div>
</DocumentFragment>
`;
